skietbaan_fe/src/components/ViewComp.js

```diff
@@ -1,130 +1,363 @@
-  }
-  onClick() {
-    history.push("/create");
-  }
-  // The method that mounts everytime there is an action detected
-  componentDidMount() {
-    this.props.fetchcomp();
-    this.props.fetchParticipants();
-  }
-  componentWillReceiveProps(val) {
-    for (let i = 0; i < val.length; i++) {
-      this.state.status.push(val.compOBJ[i].status);
-      if (i == this.state.dict.Id) {
-        this.state.dict.push(val.compOBJ[i].dict);
-      }
+	constructor(props) {
+		super(props);
+		this.state = {
+			allcomps: [],
+			status: [],
+			updatedComp: {},
+			participants: [],
+			bronzeAccuracy: '',
+			bronzeTotal: '',
+			silverAccuracy: '',
+			silverTotal: '',
+			goldAccuracy: '',
+			goldTotal: '',
+			dict: {}
+		};
+
+		this.changeStatus = this.changeStatus.bind(this);
+		this.getRequirements = this.getRequirements.bind(this);
+		this.onChange = this.onChange.bind(this);
+	}
+	// The method that mounts everytime there is an action detected
+	componentDidMount() {
+		this.props.fetchcomp();
+		this.props.fetchParticipants();
+	}
+
+	componentWillReceiveProps(val) {
+		for (let i = 0; i < val.length; i++) {
+			this.state.status.push(val.compOBJ[i].status);
+			this.state.requirements.push(val.compOBJ[i].requirements);
+			if (i == this.state.dict.Id) {
+				this.state.dict.push(val.compOBJ[i].dict);
+			}
+		}
+	}
+	//The method that detects the current status and perform the changes
+	changeStatus(status, i) {
+		const newCompOBJArr = [ ...this.props.compOBJ ]; //cloning an array of competition
+		newCompOBJArr[i].status = !status; //changing the status of a newly cloned array
+		this.props.updateByIdComp(newCompOBJArr[i], i + 1); //updating the competition status by ID in the back-end
+		const newDict = { ...this.props.dict };
+		this.setState({ compOBJ: newCompOBJArr });
+		this.setState({ dict: newDict });
+	}
+	//the method that adds the update for the inputs
+	HandleOnSubmit = (index) => {
+		const BronzeData = {
+			standard: 'Bronze',
+			accuracy: this.state.bronzeAccuracy,
+			total: this.state.bronzeTotal
+		};
+		const SilverData = {
+			standard: 'Silver',
+			accuracy: this.state.silverAccuracy,
+			total: this.state.silverTotal
+		};
+		const GoldData = {
+			standard: 'Gold',
+			accuracy: this.state.goldAccuracy,
+			total: this.state.goldTotal
+		};
+		const RData = [ BronzeData, SilverData, GoldData ];
+		this.props.UpdateRequirements(index, RData);
+	};
+	//The method that fetches the requirements everytime the competition is clicked on
+	//End-point: URL/R/{ID}
+	getRequirements(index) {
+		fetch(URL + '/R/' + index).then((response) => response.json()).then((requirementsData) =>{
+			this.setState({
+				bronzeAccuracy: requirementsData[0].accuracy,
+				bronzeTotal: requirementsData[0].total,
+				silverAccuracy: requirementsData[1].accuracy,
+				silverTotal: requirementsData[1].total,
+				goldAccuracy: requirementsData[1].accuracy,
+				goldTotal: requirementsData[2].total
```

Comment:
Should this not be goldAccuracy: requirementsData[2].accuracy

Reply:
Thanks, resolved. Wii PR just now